### ISSUE TITLE
Gapless

### ISSUE THREAD

**Issue by moechofe** [audiocogs/aurora.js#85]
What is the good implementation to play the next media without a gap ?

**Comment by Carpetsmoker** [comment]
aurora.js starts decoding on `play()`; what seems required is a `prepare_for_play()`, which would be called a few seconds before the end of the current track. This will initialise the decoder & decode the first bytes, so that `play()` can now start playing almost instantaneously.

In other words, as far as I can see it's not possible at the moment ... The good news is that it's also not too difficult to add to aurora.js, but I didn't investigate too deeply (since I can't use it now anyway due to another issue).

Your only other option is to guess the delay (ie. 200ms), and start playing the second `Player` 200ms before the end. It should be obvious that this is flawed, since you can't reliably guess this number.

**Comment by fsbdev** [comment]
Have you tried the `preload` method? https://github.com/audiocogs/aurora.js/wiki/Player-Class-Reference#preload

**Comment by Carpetsmoker** [comment]
Yeah, and I verified the files are being loaded in the network tab ... There is still at least a ~200-300ms delay (perhaps more).

**Comment by fsbdev** [comment]
Does calling `play` then immediately `pause` before actually wanting to play, then calling `play` again when ready work? I could look at the code to check but don't have the opportunity right at the moment.

**Comment by Carpetsmoker** [comment]
That's a good idea, and one of the things I tried yesterday :-) with no notable difference unfortunately...

I didn't investigate the code too much, and the above is what looks like the problem based on *glancing* the code; but I'm not absolutely sure ...
